src/oc_report_buffer.erl

```diff
@@ -0,0 +1,114 @@
+%%%------------------------------------------------------------------------
+%% Copyright 2017, OpenCensus Authors
+%% Licensed under the Apache License, Version 2.0 (the "License");
+%% you may not use this file except in compliance with the License.
+%% You may obtain a copy of the License at
+%%
+%% http://www.apache.org/licenses/LICENSE-2.0
+%%
+%% Unless required by applicable law or agreed to in writing, software
+%% distributed under the License is distributed on an "AS IS" BASIS,
+%% WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+%% See the License for the specific language governing permissions and
+%% limitations under the License.
+%%
+%% @doc Buffer of trace spans to be reported.
+%% @end
+%%%-----------------------------------------------------------------------
+-module(oc_report_buffer).
+
+-behaviour(gen_server).
+
+-export([start_link/0,
+         store_span/1]).
+
+-export([init/1,
+         handle_call/3,
+         handle_cast/2,
+         handle_info/2,
+         terminate/2]).
+
+-include("opencensus.hrl").
+
+-record(state, {reporter :: module(),
+                reporter_config :: #{},
+                send_interval_ms :: integer(),
+                timer_ref :: reference()}).
+
+-define(BUFFER_1, oc_report_buffer1).
+-define(BUFFER_2, oc_report_buffer2).
+-define(BUFFER_STATUS, oc_report_status).
+
+start_link() ->
+    maybe_init_ets(),
+    gen_server:start_link({local, ?MODULE}, ?MODULE, [], []).
+
+-spec store_span(opencensus:span()) -> ok | {error, invalid_span}.
+store_span(Span=#span{}) ->
+    [{_, Buffer}] = ets:lookup(?BUFFER_STATUS, current_buffer),
+    ets:insert(Buffer, Span);
+store_span(_) ->
+    {error, invalid_span}.
+
+init(_Args) ->
+    SendInterval = application:get_env(opencensus, send_interval_ms, 500),
+    Reporter = application:get_env(opencensus, reporter, undefined),
+    ReporterConfig = Reporter:init(),
+    Ref = erlang:send_after(SendInterval, self(), report_spans),
+    {ok, #state{reporter=Reporter,
+                reporter_config=ReporterConfig,
+                send_interval_ms=SendInterval,
+                timer_ref=Ref}}.
+
+handle_call(_, _From, State) ->
+    {noreply, State}.
+
+handle_cast(_, State) ->
+    {noreply, State}.
+
+handle_info(report_spans, State=#state{reporter=Reporter,
+                                       reporter_config=Config,
+                                       send_interval_ms=SendInterval,
+                                       timer_ref=Ref}) ->
+    erlang:cancel_timer(Ref),
+    send_spans(Reporter, Config),
+    Ref1 = erlang:send_after(SendInterval, self(), report_spans),
```

Comment:
This approach will lead to stuttering and inprecice delays between invocations so spread it out uneven. I'd like to propose an alternate approach:
```
     Ref1 = new_timer(SendInterval),
     erlang:cancel_timer(Ref),
     send_spans(Reporter, Config),
 ...

new_timer(SendInterval) ->
  receive 
    report_spans -> new_timer(SendInterval)
  after
    0-> erlang:send_after(SendInterval, self(), report_spans)
end.
```

This would keep timers as close to the Interval as possible while still not stacking them up (ps code is written outside of an editor so it is probably not compiling ;)